src/main.js

```diff
@@ -0,0 +1,269 @@
+        appContent.style.webkitTransformOrigin = origin;
+
+        console.log(`screen size: ${screenW} ${screenH} scale: ${scaleFactor}`)
+    }
+
+    function enableStyle(elementOrSelector, cssStyle, enabled) {
+        let element = (typeof elementOrSelector == 'string')
+            ? document.querySelector(elementOrSelector) : elementOrSelector;
+        if (enabled) {
+            element.classList.add(cssStyle);
+        } else {
+            element.classList.remove(cssStyle);
+        }
+    }
+
+    function renderHomePage() {
+        const homePage = document.querySelector('#home-page');
+
+        const titleDiv = homePage.querySelector('.title');
+        titleDiv.innerText = currentVideoStream.title;
+
+        const descriptionDiv = homePage.querySelector('.description');
+        descriptionDiv.innerText = currentVideoStream.description;
+
+        const tray = homePage.querySelector('.tray');
+        const selectedTile = tray.querySelector('.selected-tile');
+        selectedTile.src = currentVideoStream.cover;
+
+        setFocus('.play-content-button', () => {
+            videoController.currVideoTime = 0; // restart the video
+            showPage('playback-page')
+        });
+    }
+
+    function renderPlaybackPage() {
+        videoController.startVideo(currentVideoStream);
+
+        setFocus(videoController.video, null, action => {
+            if (action == inputActions.select || action == inputActions.playPause) {
+                videoController.togglePlayPause();
+                return true; // handled
+            }
+
+            if (action == inputActions.fastForward || action == inputActions.moveRight
+                || action == inputActions.rightShoulder1 || action == inputActions.rightShoulder2) {
+                videoController.stepForward();
+                return true; // handled
+            }
+
+            if (action == inputActions.rewind || action == inputActions.moveLeft
+                || action == inputActions.leftShoulder1 || action == inputActions.leftShoulder2) {
+                videoController.stepBackward();
+                return true; // handled
+            }
+
+            if (action == inputActions.num2 || action == inputActions.rightStick) {
+                // QA helper to allow ads to be skipped.
+                videoController.skipAd();
+                return true; // handled
+            }
+        });
+    }
+
+    function newFocusable(elementRef, selectAction, inputAction) {
+        return new Focusable(elementRef, selectAction, inputAction, focusManager);
+    }
+
+    function setFocus(elementRef, selectAction, inputAction) {
+        focusManager.setContentFocusables([newFocusable(elementRef, selectAction, inputAction)]);
+    }
+
+    function onBackAction(event) {
+        // Since the true[X] ad renderer also needs to field this event, we need to ignore when the user
+        // backs out of the ad overlay.
```

Comment:
Does this deserve a mention in the readme?

Reply:
Yes